src/EPR.Payment.Service.Data.UnitTests/Repositories/PaymentsRepositoryTests.cs

```diff
@@ -0,0 +1,199 @@
+﻿using AutoFixture.MSTest;
+using EPR.Payment.Service.Common.Data;
+using EPR.Payment.Service.Common.Data.Repositories;
+using EPR.Payment.Service.Common.UnitTests.Mocks;
+using EPR.Payment.Service.Common.UnitTests.TestHelpers;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Moq.EntityFrameworkCore;
+using System.Data.Entity;
+
+namespace EPR.Payment.Service.Data.UnitTests.Repositories
+{
+    [TestClass]
+    public class PaymentsRepositoryTests
+    {
+        private Mock<DbSet<Common.Data.DataModels.Payment>> _paymentMock;
+        private Mock<DbSet<Common.Data.DataModels.Lookups.PaymentStatus>> _paymentStatusMock;
+        public PaymentsRepositoryTests()
+        {
+            _paymentMock = MockIPaymentRepository.GetPaymentMock();
+            _paymentStatusMock = MockIPaymentRepository.GetPaymentStatusMock(true);
+        }
+
+        [TestMethod]
+        [AutoMoqData]
+        public async Task InsertPaymentStatusAsync_ValidInput_ShouldReturnGuid(
```

Comment:
There is no test case when the user tries to insert a record with same userid and organisationid. In this case the number of records should be more than one. The assertion could be that if the user is inserting a second record then the count of number of records (userid and organisationid combination) should be 2. This is a valid scenario as in the case of registration journey, a producer may try to pay fees for additional subsidiaries AFTER they have paid the fees during registration.

Reply:
resolved